src/rpft/parsers/creation/contentindexparser.py

```diff
@@ -61,49 +61,81 @@ def process_content_index_table(
                 elif row.type == "create_campaign":
                     if not len(row.sheet_name) == 1:
                         LOGGER.critical(
-                            "For create_campaign rows, exactly one sheet_name has to be specified"
+                            "For create_campaign rows, exactly one "
+                            "sheet_name has to be specified"
                         )
-                    campaign_parser = self.create_campaign_parser(sheet_reader, row)
-                    self.campaign_parsers.append((logging_prefix, campaign_parser))
+                    campaign_parser = self.create_campaign_parser(row)
+                    name = campaign_parser.campaign.name
+                    if name in self.campaign_parsers:
+                        LOGGER.warning(
+                            f"Duplicate campaign definition sheet '{name}'. "
+                            "Overwriting previous definition."
+                        )                        
+                    self.campaign_parsers[name] = (logging_prefix, campaign_parser)
                 else:
-                    LOGGER.error(f'invalid type: "{row.type}"')
+                    LOGGER.error(f"invalid type: '{row.type}'")
+
+    def _add_template(self, row, update_duplicates=False):
+        sheet_name = row.sheet_name[0]
+        # new_name = row.new_name or sheet_name
+        if sheet_name in self.template_sheets and update_duplicates:
+            LOGGER.info(
+                f"Duplicate template definition sheet '{sheet_name}'. "
+                "Overwriting previous definition."
+            )
+        if sheet_name not in self.template_sheets or update_duplicates:
+            sheet = self._get_sheet_or_die(sheet_name)
+            self.template_sheets[sheet_name] = TemplateSheet(
+                sheet, row.template_argument_definitions
+            )
+
+    def _populate_missing_templates(self):
+        for logging_prefix, row in self.flow_definition_rows:
+            with logging_context(f"{logging_prefix} | {row.sheet_name[0]}"):
+                self._add_template(row)
 
-    def process_data_sheet(self, sheet_reader, sheet_names, new_name, data_model_name):
+    def _get_sheet_or_die(self, sheet_name):
+        sheet, warnings = self.sheet_reader.get_sheet(sheet_name)
+        if sheet is None:
+            LOGGER.critical(f"Sheet {sheet_name} does not exist")
+        for warning in warnings:
+            LOGGER.warning(warning)
+        return sheet
+
+    def _process_data_sheet(self, sheet_names, new_name, data_model_name):
         if not hasattr(self, "user_models_module"):
             LOGGER.critical(
-                "If there are data sheets, a user_data_model_module_name has to be provided (as commandline argument)"
+                "If there are data sheets, a user_data_model_module_name "
+                "has to be provided (as commandline argument)"
```

Comment:
This is a library too, so command line arguments may not be relevant.

Reply:
Done.